lib/bamboo/api_error.ex

```diff
@@ -0,0 +1,34 @@
+defmodule Bamboo.ApiError do
+  @moduledoc """
+  Error used to represent a problem when sending emails through an external email service API.
+  """
+
+  defexception [:message]
+
+  def raise_api_error(message), do: raise(__MODULE__, message: message)
+  def raise_api_error(adapter, response, params, extra_message \\ "") when is_atom(adapter) do
+    message = """
+    There was a problem sending the email through the #{adapter_name(adapter)} API.
```

Comment:
I may be the only one who thinks this but it seems weird to pass in the module itself and call a function back on it here. What do you think about just passing in the name directly? With this change, the `ApiError` now has knowledge of a function on the module passed in which seems like information it shouldn't really care about.